vernac/auto_ind_decl.ml

```diff
@@ -232,9 +233,17 @@ let build_beq_scheme mode kn =
         | Lambda _-> raise (EqUnknown "abstraction")
         | LetIn _ -> raise (EqUnknown "let-in")
         | Const (kn, u) ->
-	    (match Environ.constant_opt_value_in env (kn, u) with
-	      | None -> raise (ParameterWithoutEquality (ConstRef kn))
-              | Some c -> aux (Term.applist (c,a)))
+             (match Environ.constant_opt_value_in env (kn, u) with
+              | Some c -> aux (Term.applist (c,a))
+              | None ->
+                 (* Support for working in a context with "eq_x : x -> x -> bool" *)
+                 (* Needs Hints, see test suite *)
+                 let mp,dir,lbl = Constant.repr3 kn in
+                 let kneq = Constant.make3 mp dir (Label.make ("eq_"^(Label.to_string lbl))) in
```

Comment:
This look quite ad-hoc. A principled approach would be definitely better but for now I guess this is good, as other parts of the code rely on the same kind of trick.

Reply:
Why not use change_label? It's still hacky but less adhoc.

Reply:
I was complaining about the hardwired name scheme, not the way the name was obtained. But you're right.